docs/coin_lifecycle.md

```diff
@@ -0,0 +1,119 @@
+
+## Spend Bundles
+
+Alright, so you've received some XCH and you want to deploy your first smart contract.  To do that, you're first going to have to create a **spend bundle**.  A spend bundle is a simple construct.  It is an object that contains exactly two things: a list of coin spends and an aggregated signature.
+
+```
+class SpendBundle:
+  coin_solutions: List[CoinSpend]
+  aggregated_signature: G2Element
+```
+
+A **coin spend** contains exactly three things: The coin you are trying to spend (parent_coin_info, amount, puzzle_hash), the puzzle reveal (needs to tree hash to the puzzle hash), and the solution.
+
+```
+class CoinSolution:
+  coin: Coin
+  puzzle_reveal: SerializedProgram
+  solution: SerializedProgram
+```
+
+The **aggregated signature** is a BLS signature and needs to be a signature of exactly the public key/message pairs that are output by the conditions from puzzles in the bundle.  You cannot aggregate extra signatures, nor can you leave out any.  If there are no AGG_SIG_ME or AGG_SIG_UNSAFE conditions that are output, the aggregated signature must be an empty signature which is a "c" followed 191 zeroes.
+
+Once you have constructed a spend bundle, you can push it to the `/push_tx` RPC endpoint on a Chia full node.  Keep in mind a few things though:
+
+* Unless you farm the block, you cannot guarantee if or when your spend bundle will make it to the network.  If two spends rely on each other some way, they must be in the same spend bundle and bound with signatures.
+* When using short absolute timelocks, make sure to leave in a reasonable amount of time for the bundle to reach the network.  Sometimes, due to high fee pressure, it may take a while for a low fee transaction to get in.  If your spend relies on having the coin available on the blockchain before the timelock passes, it's not a guarantee, so make sure to leave in a good amount of time just in case.
+* Nothing technically keeps the spends in a spend bundle together except for the aggregated signature and announcement logic.  If you make a spend that does not assert an announcement from another coin in the spend AND does not require a signature, that spend can be excluded from the bundle by a malicious node or farmer.
+
+Remember that other nodes are handling this piece of data as well, and it's best to assume that they will try to change everything that they can.  Make sure that if anything changes in a way you don't like, the whole bundle will fail.
```

Comment:
Maybe add that your spend bundle can be combined by others with other spend bundles as well, without your permission